src/gazebo_ros_diffdrive_uos.cpp

```diff
@@ -67,41 +56,108 @@ void GazeboRosDiffdrive::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf )
-  js_.velocity.resize(NUM_JOINTS);
-  js_.effort.resize(NUM_JOINTS);
+  if (!ros::isInitialized())
+  {
+    int argc = 0;
+    char** argv = NULL;
+    ros::init(argc, argv, "gazebo_ros_diffdrive_uos", ros::init_options::NoSigintHandler|ros::init_options::AnonymousName);
+  }
+
+  rosnode_ = new ros::NodeHandle(node_namespace_);
 
-  for (size_t i = 0; i < NUM_JOINTS; ++i)
+  cmd_vel_sub_ = rosnode_->subscribe(cmd_vel_topic_name_, 1, &GazeboRosDiffdrive::OnCmdVel, this);
+  odom_pub_ = rosnode_->advertise<nav_msgs::Odometry> (odom_topic_name_, 1);
+  joint_state_pub_ = rosnode_->advertise<sensor_msgs::JointState> (joint_states_topic_name_, 1);
+
+  enum
   {
-    js_.position[i] = 0;
-    js_.velocity[i] = 0;
-    js_.effort[i] = 0;
+    LF = 0, // left front wheel
+    LM = 1, // left middle wheel
+    LR = 2, // left rear wheel
+    RF = 3, // right front wheel
+    RM = 4, // right middle wheel
+    RR = 5  // right rear wheel
+  };
+
+  std::vector<std::string> joint_names(6);
+
+  joint_names[LF] = "left_front_wheel_joint";
+  joint_names[LM] = "left_middle_wheel_joint";
+  joint_names[LR] = "left_rear_wheel_joint";
+  joint_names[RF] = "right_front_wheel_joint",
+  joint_names[RM] = "right_middle_wheel_joint";
+  joint_names[RR] = "right_rear_wheel_joint";
+
+  std::vector<physics::JointPtr> joints_tmp(6);
+
+  for(size_t i=0; i<joint_names.size(); i++)
+  {
+    // overwrite names, if configured in sdf
+    if(_sdf->HasElement(joint_names[i]))
+    {
+      joint_names[i] = _sdf->GetElement(joint_names[i])->Get<std::string>();
+    }
+
+    // get all joint objects, some of them could not exist in the model
+    joints_tmp[i] = my_parent_->GetJoint(joint_names[i]);
   }
 
-  js_.name[0] = "left_front_wheel_joint";
-  if (_sdf->HasElement("left_front_wheel_joint"))
-    js_.name[0] = _sdf->GetElement("left_front_wheel_joint")->Get<std::string>();
+  //check for availability
 
-  js_.name[1] = "left_middle_wheel_joint";
-  if (_sdf->HasElement("left_middle_wheel_joint"))
-    js_.name[1] = _sdf->GetElement("left_middle_wheel_joint")->Get<std::string>();
+  std::vector<int> missing_joints;
 
-  js_.name[2] = "left_rear_wheel_joint";
-  if (_sdf->HasElement("left_rear_wheel_joint"))
-    js_.name[2] = _sdf->GetElement("left_rear_wheel_joint")->Get<std::string>();
+  // left front, rear and right front and rear are required for a four wheel diffdrive robot
+  if(not joints_tmp[LF]) missing_joints.push_back(LF);
+  if(not joints_tmp[LR]) missing_joints.push_back(LR);
+  if(not joints_tmp[RF]) missing_joints.push_back(RF);
+  if(not joints_tmp[RR]) missing_joints.push_back(RR);
+
+  // if one of the middle joints is available than the other should be available, too.
+  if( ((bool)joints_tmp[LM]) xor ((bool)joints_tmp[RM]))
+  {
+    missing_joints.push_back(LM);
+    missing_joints.push_back(RM);
+  }
+
+  // build helpful error message with the missing joints
+  std::string missing_err = "The controller couldn't get the joint:";
```

Comment:
there should be a trailing space after `:`, and apparently it should be `joint*s*` here.